### PR TITLE
docs: add Linux capabilities config for pyroscope java

### PR DESCRIPTION
**Pull request by marcsanmi**
As per https://github.com/grafana/alloy/pull/1788#issuecomment-2385901229

### CODE REVIEW

**Comment by jdbaldry** [comment]
Would the team consider backporting this to the latest release? I'd love to confirm that the backport workflow, most recently modified in https://github.com/grafana/pyroscope/pull/3585, is behaving as intended.

File: docs/sources/configure-client/grafana-agent/java/_index.md [hunk 1]
```diff
@@ -78,6 +78,27 @@ see [profiler-options](https://github.com/async-profiler/async-profiler?tab=read
 You must run the collector, either Grafana Alloy (recommended) or Agent (legacy), as root and inside host `pid` namespace for the `pyroscope.java`
 and `discover.process` components to work.
 
+### Additional Configuration for Linux Capabilities
```
Comment (jdbaldry) [inline docs/sources/configure-client/grafana-agent/java/_index.md:81]: We use sentence casing for headings

```suggestion
### Additional configuration for Linux capabilities
```

File: docs/sources/configure-client/grafana-agent/java/_index.md [hunk 2]
```diff
@@ -78,6 +78,27 @@ see [profiler-options](https://github.com/async-profiler/async-profiler?tab=read
...
+  securityContext:
+    runAsUser: 0
+    runAsNonRoot: false
+    capabilities:
+      add:
+        - PERFMON
+        - SYS_PTRACE
+        - SYS_RESOURCE
+        - SYS_ADMIN
+```
```
Comment (jdbaldry) [inline docs/sources/configure-client/grafana-agent/java/_index.md:95]: Technical question: If we are setting `runAsUser: 0` and `runAsNonRoot: false`, does the process not already have complete capabilities as the process is running as the root user?

Comment (marcsanmi) [reply docs/sources/configure-client/grafana-agent/java/_index.md:95]: For context, this comes from https://github.com/grafana/alloy/issues/1616.

In Kubernetes, `runAsUser: 0` and `runAsNonRoot: false` run the container as root. However, root privileges don't automatically grant all capabilities in containerized environments. The capabilities setting adds specific capabilities, but its exact behavior depends on the Container Runtime Interface (CRI) used.

AFAIK, in Docker, "The operator can whitelist specific capabilities, and any capabilities not whitelisted are dropped from the container." However, other CRIs may behave differently... 🤔 

Maybe adding a comment such as:
```yaml
# Note: Capability behavior depends on CRI settings. In Docker, non-whitelisted capabilities are dropped by default.
```

Comment (jdbaldry) [reply docs/sources/configure-client/grafana-agent/java/_index.md:95]: Thanks for explaining :)

File: examples/grafana-agent-auto-instrumentation/java/README.md [hunk 3]
```diff
@@ -40,6 +40,27 @@ After the container is operational, the Grafana Agent profiles the Java applicat
...
+        - PERFMON
+        - SYS_PTRACE
+        - SYS_RESOURCE
+        - SYS_ADMIN
+```
+These capabilities enable Alloy to access performance monitoring subsystems, trace processes, override resource limits, and perform necessary system administration tasks for profiling.
+{{< admonition type="note" >}}
+Adjust capabilities based on your specific security requirements and environment, following the principle of least privilege.
+{{< /admonition >}}
+
```
Comment (aleks-p) [inline examples/grafana-agent-auto-instrumentation/java/README.md:63]: I am not sure if this section makes sense to be in this file. We don't use Helm in this example so we can't configure this. Having it in the public docs is definitely great.

On a separate note, the `{{<admonition>}}` macros wouldn't work in these files afaik.

**Review by aleks-p — APPROVED** [review]

File: docs/sources/configure-client/grafana-agent/java/_index.md [hunk 4]
```diff
@@ -78,6 +78,27 @@ see [profiler-options](https://github.com/async-profiler/async-profiler?tab=read
 You must run the collector, either Grafana Alloy (recommended) or Agent (legacy), as root and inside host `pid` namespace for the `pyroscope.java`
 and `discover.process` components to work.
 
+### Additional configuration for Linux capabilities
```
Comment (knylander-grafana) [inline docs/sources/configure-client/grafana-agent/java/_index.md:81]:
```suggestion
### Optional:  Configure Linux capabilities
```

File: docs/sources/configure-client/grafana-agent/java/_index.md (hunk 2 above)
Comment (knylander-grafana) [reply docs/sources/configure-client/grafana-agent/java/_index.md:95]: If you choose to add the note, consider changing whitelisted to allowlisted instead.

File: examples/grafana-agent-auto-instrumentation/java/README.md [hunk 5]
```diff
@@ -40,6 +40,27 @@ After the container is operational, the Grafana Agent profiles the Java applicat
 
 You need root privileges to run the Grafana Agent for profiling. The Agent must be executed within the host's PID namespace.
 
+## Additional Configuration for Linux Capabilities
```
Comment (knylander-grafana) [inline examples/grafana-agent-auto-instrumentation/java/README.md:43]:
```suggestion
## Optional: Configure Linux capabilities
```

File: examples/grafana-agent-auto-instrumentation/java/README.md [hunk 6]
```diff
@@ -40,6 +40,27 @@ After the container is operational, the Grafana Agent profiles the Java applicat
...
+      add:
+        - PERFMON
+        - SYS_PTRACE
+        - SYS_RESOURCE
+        - SYS_ADMIN
+```
+These capabilities enable Alloy to access performance monitoring subsystems, trace processes, override resource limits, and perform necessary system administration tasks for profiling.
+{{< admonition type="note" >}}
+Adjust capabilities based on your specific security requirements and environment, following the principle of least privilege.
+{{< /admonition >}}
```
Comment (knylander-grafana) [inline examples/grafana-agent-auto-instrumentation/java/README.md:62]: Change Hugo shortcode to GitHub flavored Markdown admonition. 

```suggestion
> [!NOTE]
> Adjust capabilities based on your specific security requirements and environment, following the principle of least privilege.
```

File: examples/grafana-agent-auto-instrumentation/java/README.md (hunk 3 above)
Comment (knylander-grafana) [reply examples/grafana-agent-auto-instrumentation/java/README.md:63]: I've made a suggestion to change the admonition shortcode to one used by GitHub markdown.

**Review by knylander-grafana — APPROVED** [review]
Approving changes, pending your best opinions for where the content should be included.

**Comment by marcsanmi** [comment]
As suggested, I'm closing it to avoid confusion :)

**Pull request closed without merging**